rsocket-core/src/main/java/io/rsocket/RSocketServer.java

```diff
@@ -345,11 +348,28 @@ private void handleFireAndForget(int streamId, Mono<Void> result) {
   private void handleRequestResponse(int streamId, Mono<Payload> response) {
     response
         .doOnSubscribe(subscription -> sendingSubscriptions.put(streamId, subscription))
-        .map(payload -> PayloadFrameFlyweight.encodeNextComplete(allocator, streamId, payload))
+        .map(
+            payload -> {
+              ByteBuf byteBuf = null;
+              try {
+                byteBuf = PayloadFrameFlyweight.encodeNextComplete(allocator, streamId, payload);
+              } catch (Throwable t) {
+                if (byteBuf != null) {
+                  ReferenceCountUtil.safeRelease(byteBuf);
+                  ReferenceCountUtil.safeRelease(payload);
```

Comment:
Why this release as well as line 362?

Reply:
I  can probably remove the payload one. I might have added it when I was chasing done some memory leaks.